### PR TITLE
interface: Add crate details

### PR DESCRIPTION
**Pull request by febo**
### Problem

The current `README.md` file for the interface crate does not contain relevant information.

### Solution

Add details about the interface crate, including how to install it and examples.

### CODE REVIEW

File: interface/README.md [hunk 1]
```diff
@@ -1,3 +1,104 @@
...
+
+Creating an account:
+
+```rust
+use solana_rpc_client::rpc_client::RpcClient;
+use solana_sdk::{
+    signature::{Keypair, Signer},
+    transaction::Transaction,
+};
+use solana_system_interface::{instruction};
```
Comment (kevinheavey) [inline interface/README.md:33]: unnecessary braces around `instruction`

File: interface/README.md [hunk 2]
```diff
@@ -1,3 +1,104 @@
...
+    signature::{Keypair, Signer},
+    transaction::Transaction,
+};
+use solana_system_interface::instruction;
+use anyhow::Result;
+
+fn transfer(
+    client: &RpcClient,
+    from: &Keypair,
+    recipient: &Keypair,
```
Comment (joncinque) [inline interface/README.md:81]: nit: this should probably be a `Pubkey`

File: interface/Cargo.toml [hunk 3]
```diff
@@ -1,11 +1,13 @@
 [package]
 name = "solana-system-interface"
-version = "0.0.1"
+version = "1.0.0"
 description = "Instructions and constructors for the System program"
+authors = ["Anza Maintainers <maintainers@anza.xyz>"]
 repository = "https://github.com/solana-program/system"
+homepage = "https://anza.xyz/"
+license = "Apache-2.0"
 edition = "2021"
```
Comment (joncinque) [inline interface/Cargo.toml:9]: nit: doesn't need to be done in this PR, but how about defining all these at the workspace level and using that, ie:

```
authors = { workspace = true }
repository = { workspace = true }
homepage = { workspace = true }
license = { workspace = true }
edition = { workspace = true }
```

**Review by joncinque — APPROVED** [review]
Just a nit on one of the examples and the toml, but nothing stopping this from going in

File: interface/Cargo.toml (hunk 3 above)
Comment (febo) [reply interface/Cargo.toml:9]: Good call. I will add them here.

**Pull request merged**